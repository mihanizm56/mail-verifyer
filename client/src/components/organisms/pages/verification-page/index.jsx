import React from "react";

export const VerificationPage = () => (
	<div className="main-page">
		<h1>Verification Page</h1>
	</div>
);
