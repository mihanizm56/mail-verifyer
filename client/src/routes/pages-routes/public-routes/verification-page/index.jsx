import React from "react";
import { Route } from "react-router-dom";
import { VerificationPage } from "../../../../components";

export const VerificationPageRoute = props => <Route component={VerificationPage} {...props} />;
