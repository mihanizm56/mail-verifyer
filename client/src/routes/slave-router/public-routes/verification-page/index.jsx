import React from "react";
import { Route } from "react-router-dom";
import { VerificationPage } from "../../../../components";

export const VerificationPageRoute = () => <Route component={VerificationPage} />;
