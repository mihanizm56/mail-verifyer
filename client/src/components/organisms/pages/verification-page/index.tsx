import React, { memo } from "react";
import { Validator } from "../../../../modules";
import "./verification-page.css";

interface IRouterWithToken {
	match?: {
		params?: {
			token?: string,
		},
	};
}

export const VerificationPage = memo((props: IRouterWithToken) => {
	const { match: { params: { token = "" } = {} } = {} } = props;

	return (
		<div className="verification-page">
			<Validator token={token} />
		</div>
	);
});
