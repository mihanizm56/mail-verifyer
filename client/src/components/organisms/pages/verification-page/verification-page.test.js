import React, { createElement } from "react";
import { shallow } from "enzyme";
import { VerificationPage } from "../verification-page";

describe("test VerificationPage", () => {
	describe("test render contract", () => {
		it("render snap with all props", () => {
			const props = {
				match: { params: { token: "test token" } },
			};
			const component = shallow(<VerificationPage {...props} />);

			expect(component).toMatchSnapshot();
		});
		it("render Validator module", () => {
			const props = {
				match: { params: { token: "test token" } },
			};
			const component = shallow(<VerificationPage {...props} />);
			const Validator = component.find("Validator");

			expect(Validator.length).toEqual(1);
		});
	});
	describe("test props contract", () => {
		it("set token prop to the Validator", () => {
			const props = {
				match: {
					params: { token: "test token" },
				},
			};
			const component = shallow(<VerificationPage {...props} />);
			const Validator = component.find("Validator");
			const tokenProp = Validator.prop("token");

			expect(tokenProp).toEqual(props.match.params.token);
		});
		it("set token prop to the Validator as default string", () => {
			const component = shallow(<VerificationPage />);
			const Validator = component.find("Validator");
			const tokenProp = Validator.prop("token");

			expect(tokenProp).toEqual("");
		});
	});
});
