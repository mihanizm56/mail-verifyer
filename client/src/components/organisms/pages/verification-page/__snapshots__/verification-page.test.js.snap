// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test VerificationPage test render contract render snap with all props 1`] = `
<div
  className="verification-page"
>
  <Validator
    token="test token"
  />
</div>
`;
